src/python/pants/backend/python/goals/package_pex_binary_integration_test.py

```diff
@@ -456,3 +457,47 @@ def test_extra_build_args(rule_runner: PythonRuleRunner) -> None:
 
     assert additional_args[-2] == "--example-extra-arg"
     assert additional_args[-1] == "value-goes-here"
+
+
+def test_package_with_python_provider() -> None:
+    # Per https://github.com/pantsbuild/pants/issues/21048, test packaging a local/unconstrained pex
+    # binary when using a Python that isn't automatically visible on $PATH (using the PBS provider
+    # as just one way to get such a Python)
+
+    rule_runner = PythonRuleRunner(
+        rules=[
+            *package_pex_binary.rules(),
+            *pex_from_targets.rules(),
+            *target_types_rules.rules(),
+            *core_target_types_rules(),
+            *pbs.rules(),
+            QueryRule(BuiltPackage, [PexBinaryFieldSet]),
+        ],
+        target_types=[
+            PexBinary,
+            PythonSourcesGeneratorTarget,
+        ],
+    )
+
+    rule_runner.write_files(
+        {
+            "app.py": "",
+            "BUILD": dedent(
+                """\
+                python_sources(name="src")
+                pex_binary(name="target", entry_point="./app.py")
+                """
+            ),
+        }
+    )
+
+    tgt = rule_runner.get_target(Address("", target_name="target"))
+    field_set = PexBinaryFieldSet.create(tgt)
+
+    # a random (https://xkcd.com/221/) old version of Python, that seems unlikely to be installed on
```

Comment:
We could check for this and skip or fail the test if it *is* installed, with an appropriate warning.

Reply:
Good idea... but I'm inclined to say it isn't worth the effort: as long as CI doesn't have 3.10.2, it's not super concerning if someone happens to have it installed.

Plus, on further looking, this test never calls `rule_runner.set_options([], env_inherit={"PATH", "PYENV_ROOT", "HOME"})` or similar. That is, the `rule_runner` doesn't have `PATH` (etc.) to do a global interpreter search anyway, even if such an interpreter is installed.

(As validation: I tried changing this to a version of Python I do have installed, and the test still failed, without the `--python-path` change.)

I've adjusted the comment here.

Reply:
Yeah, fine with me!